homeassistant/components/n26/__init__.py

```diff
@@ -0,0 +1,146 @@
+"""Support for N26 bank accounts."""
+from datetime import datetime, timedelta, timezone
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_PASSWORD, CONF_SCAN_INTERVAL, CONF_USERNAME)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.util import Throttle
+
+from .const import DATA, DOMAIN
+
+REQUIREMENTS = ['n26==0.2.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CARD_ID = "card_id"
+DEFAULT_SCAN_INTERVAL = timedelta(minutes=30)
+
+# define configuration parameters
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_CARD_ID): cv.string,
+        vol.Optional(CONF_SCAN_INTERVAL,
+                     default=DEFAULT_SCAN_INTERVAL): cv.time_period,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+N26_COMPONENTS = [
+    'sensor',
+    'switch'
+]
+
+
+def setup(hass, config):
+    """Set up N26 Component."""
+    user = config[DOMAIN][CONF_USERNAME]
+    password = config[DOMAIN][CONF_PASSWORD]
+    card_id = config[DOMAIN][CONF_CARD_ID]
+
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][DATA] = N26Data(user, password, card_id)
+
+    # Load components for supported devices
+    for component in N26_COMPONENTS:
+        load_platform(hass, component, DOMAIN, {}, config)
+
+    return True
+
+
+def timestamp_ms_to_date(epoch_ms) -> datetime or None:
+    """Convert millisecond timestamp to datetime."""
+    if epoch_ms:
+        return datetime.fromtimestamp(epoch_ms / 1000, timezone.utc)
+
+
+class N26Data:
+    """Handle N26 API object and limit updates."""
+
+    def __init__(self, user: str, password: str, card_id: str):
+        """Initialize the data object."""
+        from n26 import api, config as api_config
+        self._api = api.Api(api_config.Config(user, password, card_id))
+
+        self._account_info = {}
+        self._balance = {}
+        self._limits = {}
+        self._account_statuses = {}
+
+        self._cards = {}
+        self._spaces = {}
+
+        self.update()
```

Comment:
We try to avoid side effects in init method. Call this separately after instantiation.

Reply:
done